demos/widgets/lv_demo_widgets.c

```diff
@@ -1419,17 +1419,17 @@ static void chart_event_cb(lv_event_t * e)
             lv_draw_triangle_dsc_t tri_dsc;
 
             lv_draw_triangle_dsc_init(&tri_dsc);
-            tri_dsc.p[0].x = (int32_t)draw_line_dsc->p1_x;
-            tri_dsc.p[0].y = (int32_t)draw_line_dsc->p1_y;
-            tri_dsc.p[1].x = (int32_t)draw_line_dsc->p2_x;
-            tri_dsc.p[1].y = (int32_t)draw_line_dsc->p2_y;
-            tri_dsc.p[2].x = (int32_t)(draw_line_dsc->p1_y < draw_line_dsc->p2_y ? draw_line_dsc->p1_x : draw_line_dsc->p2_x);
-            tri_dsc.p[2].y = (int32_t)LV_MAX(draw_line_dsc->p1_y, draw_line_dsc->p2_y);
+            tri_dsc.p[0].x = (int32_t)draw_line_dsc->p1.x;
```

Comment:
Let's add an API to deal with this potential type conversion.
lv_point_copy(lv_point_t, lv_point_t);
lv_point_precise_copy(lv_point_precise_t, lv_point_precise_t);

lv_point_copy_from_precise(lv_point_t, lv_point_precise_t);
lv_point_copy_to_precise(lv_point_precise_t, lv_point_t);

Reply:
Ideally we can hide the `lv_point_t` details and force user to use API to access `x`,`y`.

Reply:
> lv_point_copy(lv_point_t, lv_point_t);
lv_point_precise_copy(lv_point_precise_t, lv_point_precise_t);

Do we really need these? `p1 = p2` can do the same.

> lv_point_copy_from_precise(lv_point_t, lv_point_precise_t);
lv_point_copy_to_precise(lv_point_precise_t, lv_point_t);

I agree but  I suggest this API:
```c
lv_point_t lv_point_from_precise(lv_point_precise_t * );
lv_point_precise_t lv_point_to_precise(lv_point_t * );
```

Reply:
OK. Let's go with adding API only for mismatched types.

Reply:
My only concern is that `lv_value_precise_t` could be same as `int32_t`. Thus code written for `!LV_USE_FLOAT` could potentially misuse p1 = p2 between `lv_value_precise_t` `lv_value_t`.

And I'm certain this will happen because we have just fixed bugs when deal with `lv_coord_t` where people misuse it with int16_t.

Reply:
> My only concern is that lv_value_precise_t could be same as int32_t. Thus code written for !LV_USE_FLOAT could potentially misuse p1 = p2 between lv_value_precise_t lv_value_t.

Can we assume that issues like this will be filtered out by our CI which compiles and tests LVGL with `LV_USE_FLOAT` on and off too?

> And I'm certain this will happen because we have just fixed bugs when deal with lv_coord_t where people misuse it with int16_t.

I know it's just an example, but as you probably know `lv_coord_t` is removed and `int32_t` is used instead.

Reply:
> Can we assume that issues like this will be filtered out by our CI which compiles and tests LVGL with `LV_USE_FLOAT` on and off too?
> 
> >

Unfortunately no. The faulty code is from application code written by inexperienced engineer.
They have used int16_t instead of lv_coord_t in v8, and causes incompatibility when we removed lv_coord_t and replaced with int32_t in v9.